packages/rocketchat-ui-sidenav/client/chatRoomItem.js

```diff
@@ -34,18 +34,27 @@ Template.chatRoomItem.helpers({
 			unread,
 			active,
 			archivedClass,
-			statusClass: this.t === 'd' ? Session.get(`user_${ this.name }_status`) || 'offline' : this.t === 'l' ? RocketChat.roomTypes.getUserStatus(this.t, this.rid) || 'offline' : false
+			status: this.t === 'd' || this.t === 'l'
 		};
 
-		if (RocketChat.settings.get('Store_Last_Message')) {
-			if (this.lastMessage) {
-				roomData.lastMessage = this.lastMessage;
-			} else {
-				const room = RocketChat.models.Rooms.findOne(this.rid || this._id, { fields: { lastMessage: 1 } });
-				roomData.lastMessage = room && room.lastMessage || { msg: t('No_messages_yet') };
-			}
+		if (!this.lastMessage && RocketChat.settings.get('Store_Last_Message')) {
+			const room = RocketChat.models.Rooms.findOne(this.rid || this._id, { fields: { lastMessage: 1 } });
+			roomData.lastMessage = room && room.lastMessage || { msg: t('No_messages_yet') };
 		}
-
 		return roomData;
 	}
 });
+
+RocketChat.callbacks.add('enter-room', (sub) => {
```

Comment:
my channels did not have active indicators while testing

Reply:
fixed :)